src/vendor/vendor.c

```diff
@@ -1670,25 +1584,25 @@ int RdkResetInner(char *group_name, char *path, char *value, char *debug_msg)
     }
 
     // Fill in the details of the parameter controlling reboot or factory reset
-    rdk_param.parameterName = path;
-    rdk_param.parameterValue = value;
-    rdk_param.type = ccsp_string;
+    rbusValue_t val ;
+    rbusValue_Init(&val);
+    rbusValue_SetString(val, value);
 
     // Exit if unable to set the parameter
-    rdkc = &rdk_components[group_id];
-    ccsp_err = CcspBaseIf_setParameterValues(bus_handle, rdkc->component_name, rdkc->dbus_path, 
-                                             RDK_SESSION_ID, CCSP_USP_WRITE_ID,
-                                             &rdk_param, 1,
-                                             RDK_COMMIT_NOW, &fault_param);
-    if (ccsp_err != CCSP_SUCCESS)
-    {
-        USP_ERR_SetMessage("%s: CcspBaseIf_setParameterValues(%s) failed (%d - %s)", __FUNCTION__, path, ccsp_err, ToCcspErrString(ccsp_err));
+    rbus_err =rbus_set(bus_handle, path, val, NULL);
+    if (rbus_err != RBUS_ERROR_SUCCESS)
+    {
+        USP_ERR_SetMessage("%s:rbus_set(%s) failed (%d - %s)", __FUNCTION__, path, rbus_err, ToRbusErrString(rbus_err));
         return USP_ERR_INTERNAL_ERROR;
     }
 
     // USP Agent exits
     USP_LOG_Info("%s: Performing %s", __FUNCTION__, debug_msg);
-    exit(0);
+    rbusValue_Release(val);
+    if((strcmp(debug_msg, "Reboot")==0) || (strcmp(debug_msg, "Factory Reset")==0))
```

Comment:
can you explain about this reboot and factory reset here

Reply:
-> The functionality is when we execute "Device.Reboot()" or "Device.FactoryReset()" , the function we do exit(0) and stops the USPPA main and will do reboot operation 
-> So to avoid stopping USSPA, While performing above mentioned sample command : **Device.Diagnostic()**, added if condition for Reboot and Factory Reset

Reply:
Lets do a live review.i need some clarifications